core/imodel-bridge/src/BADGERSIssueReporter.ts

```diff
@@ -0,0 +1,288 @@
+        //If the id is not the same throw a warning
+        const values = [this._contextInfo.activityId, this._contextInfo.contextId, this._contextInfo.jobId, this._contextInfo.iModelId, this._contextInfo.briefcaseId, srcId, this._fileInfo.filePath,
+            "SourceFile", itemType, dataSource, name, exists, downloadUrl, state, failureReason, uniqueName, fileSize, foundByBridge]
+        this._reportDb.withPreparedSqliteStatement(`INSERT INTO ${this.SRCFILE_TEMP_BRIDGE_ISSUES_TABLE}(activityId, contextId, jobId, imodelId, briefcaseId, fileId, filePath, dataType, itemType, dataSource, fileName, itemExists, downloadUrl, state, failureReason, uniqueName, fileSize, foundByBridge) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)`, (stmt) => {
+            stmt.bindValues(values);
+            stmt.step();
+        })
+    }
+
+    public recordReferenceFileInfo(srcId: string, name: string, uniqueName: string, itemType: string, dataSource: string, downloadUrl: string, state: string, failureReason: string, exists: boolean, fileSize: number, foundByBridge: boolean) {
+        const values = [this._contextInfo.activityId, this._contextInfo.contextId, this._contextInfo.jobId, this._contextInfo.iModelId, this._contextInfo.briefcaseId, srcId, this._fileInfo.filePath,
+            "ReferenceFile", itemType, dataSource, name, exists, downloadUrl, state, failureReason, uniqueName, fileSize, foundByBridge]
+        this._reportDb.withPreparedSqliteStatement(`INSERT INTO ${this.SRCFILE_TEMP_BRIDGE_ISSUES_TABLE}(activityId, contextId, jobId, imodelId, briefcaseId, fileId, filePath, dataType, itemType, dataSource, fileName, itemExists, downloadUrl, state, failureReason, uniqueName, fileSize, foundByBridge) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)`, (stmt) => {
+            stmt.bindValues(values);
+            stmt.step();
+        })
+    }
+
+    public createJsonReport() {
+        const timestamp = new Date().toISOString();
+        const context = {
+            "reportType": "detailReport",
+            "operationType": this._contextInfo.operationType,
+            "jobid": this._contextInfo.jobId,
+            "contextid": this._contextInfo.contextId,
+            "imodelid": this._contextInfo.iModelId,
+            "activityid": this._contextInfo.activityId,
+            "briefcaseid": this._contextInfo.briefcaseId,
+            "timestamp": timestamp
+        };
+
+        const sourceFile: any = {};
+        this._reportDb.withPreparedSqliteStatement(`SELECT dataType, itemType, dataSource, fileId, fileName, itemExists, downloadUrl, state, failureReason, uniqueName, fileSize FROM ${this.SRCFILE_TEMP_BRIDGE_ISSUES_TABLE} WHERE activityId=? AND foundByBridge=? AND dataType=?`,
+            (stmt) => {
+                stmt.bindValue(1, this._contextInfo.activityId);
+                stmt.bindValue(2, true);
+                stmt.bindValue(3, "SourceFile");
+                while (stmt.step() === DbResult.BE_SQLITE_ROW) {
```

Comment:
In the case of multiple SourceFile records, native code seems to take the first, whereas your `while` loop will take the last... while this case is unexpected, I'd rather see parity just in case.